src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vedtak/sammensattkontrollsak/SammensattKontrollsakValidator.kt

```diff
@@ -0,0 +1,68 @@
+package no.nav.familie.ks.sak.kjerne.behandling.steg.vedtak.sammensattkontrollsak
+
+import no.nav.familie.ks.sak.common.exception.FunksjonellFeil
+import no.nav.familie.ks.sak.config.BehandlerRolle
+import no.nav.familie.ks.sak.config.featureToggle.FeatureToggleConfig
+import no.nav.familie.ks.sak.config.featureToggle.UnleashNextMedContextService
+import no.nav.familie.ks.sak.kjerne.behandling.BehandlingService
+import no.nav.familie.ks.sak.sikkerhet.TilgangService
+import org.springframework.stereotype.Component
+
+@Component
+class SammensattKontrollsakValidator(
+    private val tilgangService: TilgangService,
+    private val behandlingService: BehandlingService,
+    private val unleashService: UnleashNextMedContextService,
+) {
+    fun validerHentSammensattKontrollsakTilgang() {
+        if (!unleashService.isEnabled(FeatureToggleConfig.KAN_OPPRETTE_OG_ENDRE_SAMMENSATTE_KONTROLLSAKER)) {
+            throw FunksjonellFeil(melding = "Mangler tilgang for å hente sammensatt kontrollsak.")
+        }
+
+        tilgangService.validerTilgangTilHandling(
+            minimumBehandlerRolle = BehandlerRolle.SAKSBEHANDLER,
+            handling = "Hent SammensattKontrollsak",
+        )
+    }
+
+    fun validerOpprettSammensattKontrollsakTilgang() {
+        if (!unleashService.isEnabled(FeatureToggleConfig.KAN_OPPRETTE_OG_ENDRE_SAMMENSATTE_KONTROLLSAKER)) {
+            throw FunksjonellFeil(melding = "Mangler tilgang for å opprette sammensatt kontrollsak.")
+        }
+
+        tilgangService.validerTilgangTilHandling(
+            minimumBehandlerRolle = BehandlerRolle.SAKSBEHANDLER,
+            handling = "Opprett SammensattKontrollsak",
+        )
+    }
+
+    fun validerOppdaterSammensattKontrollsakTilgang() {
+        if (!unleashService.isEnabled(FeatureToggleConfig.KAN_OPPRETTE_OG_ENDRE_SAMMENSATTE_KONTROLLSAKER)) {
+            throw FunksjonellFeil(melding = "Mangler tilgang for å oppdatere sammensatt kontrollsak.")
+        }
+
+        tilgangService.validerTilgangTilHandling(
+            minimumBehandlerRolle = BehandlerRolle.SAKSBEHANDLER,
+            handling = "Oppdater SammensattKontrollsak",
+        )
+    }
+
+    fun validerSlettSammensattKontrollsakTilgang() {
+        if (!unleashService.isEnabled(FeatureToggleConfig.KAN_OPPRETTE_OG_ENDRE_SAMMENSATTE_KONTROLLSAKER)) {
+            throw FunksjonellFeil(melding = "Mangler tilgang for å slette sammensatt kontrollsak.")
+        }
+
+        tilgangService.validerTilgangTilHandling(
+            minimumBehandlerRolle = BehandlerRolle.SAKSBEHANDLER,
+            handling = "Slett SammensattKontrollsak",
+        )
+    }
+
+    fun validerRedigerbarBehandling(behandlingId: Long) {
+        behandlingService.hentBehandling(behandlingId).run {
+            if (status.erLåstMenIkkeAvsluttet() || erAvsluttet()) {
+                throw FunksjonellFeil("Behandlingen er låst for videre redigering ($status)")
+            }
+        }
+    }
+}
```

Comment:
Denne her kan egentlig legges et annet sted som ikke har noe med sammensatt kontroll sak å gjøre, da den er ganske generell.

Reply:
Ja, vurderte å legge inn en metode "erRedigerbar" i `Behandling` filer, samt kanskje lage en `BehandlingValidator` klasse, men var ikke helt sikker på om "erRedigerbar" sjekken var så generell at den ville fungere for alle andre klasser